### PR TITLE
ClassLoader

### PR DESCRIPTION
**Pull request by ShuFFel**

### CODE REVIEW

File: ClassLoader/pom.xml [hunk 1]
```diff
@@ -0,0 +1,25 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <groupId>MentoringProgramm</groupId>
```
Comment (mx510ss) [inline ClassLoader/pom.xml]: In most cases `groupId` =` root package name` of the project
So it should be something like com.instinctools.egor.mentoring.classloader

File: ClassLoader/src/main/java/company/Helper.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Method;
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.util.Scanner;
+
+public class Helper {
```
Comment (mx510ss) [inline ClassLoader/src/main/java/company/Helper.java]: Bad naming also - Helper say nothing for anyone except you
please rename it

File: ClassLoader/src/main/java/company/Helper.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+    Class<?> aClass = null;
+    Logger log = LogManager.getLogger(Helper.class);
+    public Helper() {
+        while (true){
+            log("Menu: \n1-start \n0-exit \n");
```
Comment (mx510ss) [inline ClassLoader/src/main/java/company/Helper.java]: Better to use some specific logging level like debug, info, error, etc.
please update it for all places

File: ClassLoader/src/main/java/company/Helper.java [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+            Integer chose = scanner.nextInt();
+            if(chose == 1) {
+                log("Input full path to jar: ");
+                String path = scanner.next();
+                log("Input jar name: ");
+                String jarName = scanner.next();
+                try {
+                    String url = "file://"+path+"/" + jarName;
+                    URL[] urlFromPath = new URL[]{new URL(url)};
+                    URLClassLoader loader = new URLClassLoader(urlFromPath);
```
Comment (mx510ss) [inline ClassLoader/src/main/java/company/Helper.java]: The Idea of this task was to use your own classloader not a standard one.
And the easiest way is to extend `URLClassLoader` by your own implementation (or just wrap it)

File: ClassLoader/src/main/java/company/Main.java [hunk 5]
```diff
@@ -0,0 +1,11 @@
+package company;
+
+public class Main {
+
+    public static void main(String[] args) {
+        System.out.println(String.class.getClassLoader());
```
Comment (mx510ss) [inline ClassLoader/src/main/java/company/Main.java]: Didn't get why we need `System.out.println` here. Please use logging or just remove these lines

File: ClassLoader/src/main/java/company/MyLoader.java [hunk 6]
```diff
@@ -0,0 +1,49 @@
+package company;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+
+public class MyLoader extends ClassLoader {
```
Comment (mx510ss) [inline ClassLoader/src/main/java/company/MyLoader.java]: This class is not used in the app. So why you created it)?

**Review by mx510ss — CHANGES_REQUESTED** [review]
Need changes + need to rename main package:
you have only one package = company
its wrong naming according to http://www.oracle.com/technetwork/java/codeconventions-135099.html
and do not make any sense. 
Your package should be something like:
com.instinctools.egor.mentoring.classloader

File: ClassLoader/src/main/java/company/Helper.java (hunk 3 above)
Comment (ShuFFel) [reply ClassLoader/src/main/java/company/Helper.java]: I already use info level

**Pull request closed without merging**